### PR TITLE
Pad out spaces on progress monitoring. 

### PR DESCRIPTION
**Pull request by hoytak**
Puts a minimum width for the line we print in our progress bar, with the difference made up with spaces.  This is based on git's messages, which can be longer than ours, causing weird interleaving when the user does stuff to the input (hits a new line, etc.)  .

### CODE REVIEW

File: rust/progress_reporting/src/data_progress.rs [hunk 1]
```diff
@@ -297,8 +304,10 @@ impl DataProgressReporter {
             }
         };
 
-        if write_str.len() < lg_print_info.last_write_length {
-            let mut len_to_write = lg_print_info.last_write_length - write_str.len();
+        let write_str_pad_len = lg_print_info.last_write_length.min(PRINT_LINE_MIN_WIDTH);
```
Comment (seanses) [inline rust/progress_reporting/src/data_progress.rs:307]: .max?

File: rust/progress_reporting/src/data_progress.rs [hunk 2]
```diff
@@ -143,6 +143,11 @@ impl DataProgressReporter {
 
     /// Does the actual printing
     fn print(&self, is_final: bool) -> std::result::Result<(), std::io::Error> {
+        // Put a minimum width for the line we print.  This is based on git's messages,
+        // which can be longer than ours, causing weird inteleaving when the user does
+        // stuff to the input (hits a new line, etc.)  So, print a minimum buffer of spaces.
+        const PRINT_LINE_MIN_WIDTH: usize = 74;
```
Comment (seanses) [inline rust/progress_reporting/src/data_progress.rs:149]: Wondering why choosing 74

File: rust/progress_reporting/src/data_progress.rs (hunk 1 above)
Comment (hoytak) [reply rust/progress_reporting/src/data_progress.rs:307]: Oh bleh.  Great catch.

File: rust/progress_reporting/src/data_progress.rs (hunk 2 above)
Comment (hoytak) [reply rust/progress_reporting/src/data_progress.rs:149]: Measured the length of the git progress strings that the original bug report showed.

**Review by seanses — APPROVED** [review]
LGTM!

**Pull request merged**